### ISSUE TITLE
Create a service importer pod

### ISSUE THREAD

**Issue by jstrachan** [fabric8io/fabric8#4630]
Folks may use multiple environments for different microservices (eg 'dev', 'qa', 'prod') but in each envirornment folks might want to import shared services (eg elasticsearch, grafana, Prometheus).

It would be nice to have an easy way to import services from namespace X into namespace Y so they can be shared across namespaces.

I wonder if we need a 'sharing' pod that we run in namespace Y that watches services in namespace X with some optional label filter and makes sure they are mirrored into namespace Y via their DNS / portalIP? Creating services in Y that reference each matching service in X

Then folks could run management stuff in namespace 'foo' and then in each project namespace 'bar' folks could easily share those services via running a share pod using an OpenShift template which just refers to the 'foo' namespace. Then as services are added/updated in 'foo' they get replicated into 'bar'.

This would help folks run shared stuff in multiple environments without duplicating processes & databases etc

**Comment by jstrachan** [comment]
We should ensure that each imported service in namespace 'Y' gets annotated by the importer - so that if the imported service in 'X' is deleted, the importer can safely delete the corresponding service in 'Y'.

Also folks in 'Y' may wish to import all services from 'X' but then choose to override some service 'foo' with a local implementation. In which case the user an replace the imported service 'foo' with a custom one.

The importer should then only import services from 'X' which don't already exist in 'Y'.

Looking in any namespace it should then be clear which services are local & which are imported (and from which namespace) via annotations

**Comment by jimmidyson** [comment]
Can probably just access it via DNS? Also look at how external services are
managed- this is the same for services across namespaces.

**Comment by jstrachan** [comment]
Yeah - am thinking the service importer just creates a 'foo' service in Y that can be accessed via env vars or DNS - which just points to the foo.X DNS name / clusterIP etc. so it looks like 'foo' service is implemented locally but it points to the X namespace as a kinda external service

**Comment by jstrachan** [comment]
The service importer idea is a possible plugin to implement something like this idea:
https://github.com/openshift/origin/issues/1244
